android/tangram/src/com/mapzen/tangram/MapController.java

```diff
@@ -123,35 +123,13 @@ private Bitmap capture() {
      * must contain all the local files that the map will need
      * @param sceneFilePath Location of the YAML scene file within the assets directory
      */
-    protected MapController(Context context, String sceneFilePath) {
+    protected MapController(GLSurfaceView view, String sceneFilePath) {
 
         scenePath = sceneFilePath;
 
-        // Get configuration info from application
-        displayMetrics = context.getResources().getDisplayMetrics();
-        assetManager = context.getAssets();
-
-        // Load the fonts
-        fontFileParser = new FontFileParser();
-        fontFileParser.parse("/system/etc/fonts.xml");
-
         // Set a default HTTPHandler
         httpHandler = new HttpHandler();
 
-        nativeInit(this, assetManager, scenePath);
-
-    }
-
-    static MapController getInstance(Context context, String sceneFilePath) {
-        return new MapController(context, sceneFilePath);
-    }
-
-    /**
-     * Set the view in which the map will be drawn
-     * @param view GLSurfaceView where the map will be displayed; input events from this view will
-     * be handled by the resulting MapController
-     */
-    void setView(GLSurfaceView view) {
         touchInput = new TouchInput(view.getContext());
```

Comment:
I thought MapView is our external API now (setView is package visible only) and GLSurfaceView is  added to the Fragment in postExecute (again) so it cant get input before complete initialiazation